### PR TITLE
[testing][e2e] Add basic tests for network modules

### PR DESCRIPTION
**Pull request by Beastlex**
## Description
Basic e2e-tests for modules:
* istio
* static-routing-manager
* cilium (EgressGateway and EgressGatewayPolicy)

## Why do we need it, and what problem does it solve?
Tests will help to detect modules deployment errors at an early stage.

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: testing
type: chore
summary: basic e2e tests for network modules
impact_level: low
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/static'...

File: testing/cloud_layouts/script.d/wait_cluster_ready/test_script.sh [hunk 1]
```diff
@@ -35,6 +35,16 @@ else
   ingress=""
 fi
 
+if [ "$(kubectl -n d8-istio get po | grep operator | wc -l)" -gt 0 ]; then
```
Comment (apolovov) [inline testing/cloud_layouts/script.d/wait_cluster_ready/test_script.sh:38]: Why operator? We need the running istiod pod.

File: testing/cloud_layouts/Static/resources.tpl.yaml [hunk 2]
```diff
@@ -62,3 +62,42 @@ spec:
...
+        dportRange:
+          start: 300
+          end: 400
+        ipProto: 6
+      actions:
+        lookup:
+          routingTableName: myrt-main
+      priority: 50
+  nodeSelector:
+    dedicated.worker: ""
```
Comment (apolovov) [inline testing/cloud_layouts/Static/resources.tpl.yaml:103]: Egressgateway with virtual ip too!

**Review by apolovov — CHANGES_REQUESTED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Beastlex. I've started the workflow for label 'e2e/run/gcp'...

**Review by apolovov — APPROVED** [review]

**Comment by z9r5** [comment]
/e2e/abort static;Static;containerd;1.27 10073762070 10073762070-1-con-1-27 /sys/deckhouse-oss/install:pr9025 astra@192.168.199.41

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort gcp;WithoutNAT;containerd;1.27 10093718612 10093718612-1-con-1-27 /sys/deckhouse-oss/install:pr9025 user@34.141.22.111

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10090387397 10090387397-1-con-1-27 /sys/deckhouse-oss/install:pr9025 redos@89.169.129.14

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10089453785 10089453785-1-con-1-27 /sys/deckhouse-oss/install:pr9025 redos@89.169.129.157

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Pull request merged**